redbox-core/redbox/graph/root.py

```diff
@@ -241,10 +240,10 @@ def get_root_graph(
     # Subgraphs
     chat_subgraph = get_chat_graph(debug=debug)
     cwd_subgraph = get_chat_with_documents_graph(
-        all_chunks_retriever=all_chunks_retriever,
+        retriever=retriever,
         debug=debug,
     )
-    metadata_subgraph = get_retrieve_metadata_graph(metadata_retriever=metadata_retriever, debug=debug)
+    metadata_subgraph = get_retrieve_metadata_graph(retriever=retriever, debug=debug)
```

Comment:
What is the metadatasubgraph? Do we need it, and how does it differ from the chat with documents subgraph?

Reply:
this is defined here https://github.com/i-dot-ai/redbox/blob/1cb033c4544372d8fc35d4135b03e647651bd73b/redbox-core/redbox/graph/root.py#L209 it is primarily used to retrieve the token "length" of the documents so that we know which strategy to use when querying docs:
* chat with docs small 
* chat with docs large (map-reduce)
* RAG (search)
* error (too large)

as we remove search and map-reduce the need for this will get simpler but we will still need to know whether the token size is too large (although we almost certainly dont need a graph to compute this)

Reply:
Thanks George. I am trying to work out why this is a subgraph rather than just a node. What am I missing?

Reply:
not sure tbh. this may be one for @jamesrichards4

Reply:
Fair. We can save it for a separate PR anyway. I will record it in the excel sheet